pkg/fanal/analyzer/language/rust/cargo/cargo.go

```diff
@@ -187,8 +192,29 @@ func (a cargoAnalyzer) parseCargoTOML(fsys fs.FS, path string) (map[string]strin
 	}
 
 	// https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#inheriting-a-dependency-from-a-workspace
-	maps.Copy(dependencies, tomlFile.Workspace["dependencies"])
+	maps.Copy(dependencies, tomlFile.Workspace.Dependencies)
+	// https://doc.rust-lang.org/cargo/reference/workspaces.html#the-members-and-exclude-fields
+	return dependencies, tomlFile.Workspace.Members, nil
+}
+
+func (a cargoAnalyzer) parseCargoTOML(fsys fs.FS, filePath string) (map[string]string, error) {
+	dependencies, members, err := tomlDependencies(fsys, filePath)
+	if err != nil {
+		return nil, err
+	}
+	// According to Cargo workspace RFC, workspaces can't be nested:
+	// https://github.com/nox/rust-rfcs/blob/master/text/1525-cargo-workspace.md#validating-a-workspace
+	for _, member := range members {
+		memberPath := path.Join(path.Join(path.Dir(filePath), member), types.CargoToml)
+		memberDeps, _, err := tomlDependencies(fsys, memberPath)
+		if err != nil {
+			log.Logger.Warnf("Unable to parse %q: %s", memberPath, err)
+			continue
+		}
+		maps.Copy(dependencies, memberDeps)
```

Comment:
IIUC, `workspace.dependencies` is overridden by mistake here.

Reply:
You are right.
Thanks!
Fixed in https://github.com/aquasecurity/trivy/pull/5285/commits/7b319754bbe713004b5d45979e514f2b1e9464ea